components/brave_wallet_ui/common/hooks/transaction-parser.ts

```diff
@@ -194,6 +195,9 @@ export function useTransactionParser (
         const { gasFeeFiat } = feeDetails
         const totalAmountFiat = Number(gasFeeFiat).toFixed(2)
         const insufficientNativeFunds = Number(gasFeeFiat) > Number(accountsNativeFiatBalance)
+        const formatedValue = formatBalance(amount, token?.decimals ?? 18)
+        const userTokenBalance = formatBalance(usersTokenInfo?.assetBalance ?? '', usersTokenInfo?.asset?.decimals ?? 18)
+        const allowanceValue = Number(formatedValue) > Number(userTokenBalance) ? getLocale('braveWalletTransactionApproveUnlimited') : userTokenBalance
```

Comment:
Isn't it safer to compare `amount` and `usersTokenInfo?.assetBalance` directly to avoid a floating-point comparison?

Reply:
Updated!